### PR TITLE
Have the AppManager set up routing for AppServers

### PR DESCRIPTION
**Pull request by cdonati**

### CODE REVIEW

**Comment by cdonati** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/1043/

File: AppServer/google/appengine/api/appcontroller_client.py [hunk 1]
```diff
@@ -312,3 +312,50 @@ def run_groomer(self):
...
     """
     return self.call(self.MAX_RETRIES, self.server.run_groomer, self.secret)
+
+
+  def add_routing_for_appserver(self, app_id, appserver_ip, port):
+    """ Tells the AppController to begin routing traffic to an AppServer.
+
+    Args:
+      app_id: A string that contains the application ID.
+      appserver_ip: A string that contains the IP Address of the instance
```
Comment (menivaitsi) [inline AppServer/google/appengine/api/appcontroller_client.py]: Address -> address

File: AppServer/google/appengine/api/appcontroller_client.py [hunk 2]
```diff
@@ -312,3 +312,50 @@ def run_groomer(self):
...
+    return self.call(self.MAX_RETRIES, self.server.add_routing_for_appserver,
+      app_id, appserver_ip, port, self.secret)
+
+
+  def remove_appserver_from_haproxy(self, app_id, appserver_ip, port):
+    """ Tells the AppController to stop routing traffic to an AppServer.
+
+    Args:
+      app_id: A string that contains the application ID.
+      appserver_ip: A string that contains the IP Address of the instance
```
Comment (menivaitsi) [inline AppServer/google/appengine/api/appcontroller_client.py]: Same here/

File: AppManager/test/unit/test_app_manager_server.py [hunk 3]
```diff
@@ -119,6 +120,7 @@ def test_start_app_goodconfig_java(self):
                         .and_return()
     flexmock(subprocess).should_receive('call')\
                         .and_return(0)
+    flexmock(app_manager_server).should_receive('add_routing')
```
Comment (menivaitsi) [inline AppManager/test/unit/test_app_manager_server.py]: Please mock the remove_routing calls as well.

File: AppController/djinn.rb [hunk 4]
```diff
@@ -5361,20 +5338,7 @@ def remove_appserver_process(app_id, port, secret)
...
-      begin
-        acc.remove_appserver_from_haproxy(app, my_node.private_ip, port)
-      rescue FailedNodeException
-        Djinn.log_warn("Failed to remove appserver from haproxy for app #{app}")
-      end
-    end
-
-    # And tell the AppDashboard that the AppServer has been killed.
+    # Tell the AppDashboard that the AppServer has been killed.
     delete_instance_from_dashboard(app, "#{my_node.private_ip}:#{port}")
```
Comment (obino) [inline AppController/djinn.rb]: Should this be done by the appmanagerserver? is there already an ability to talk to the dashboard from the python side?

**Comment by cdonati** [comment]
Precise: https://ocd.appscale.com:8080/job/Daily%20Build/1051/

**Comment by cdonati** [comment]
Precise: https://ocd.appscale.com:8080/job/Daily%20Build/1053/
Trusty: https://ocd.appscale.com:8080/job/Daily%20Build/1054/

**Comment by obino** [comment]
:+1:

**Comment by cdonati** [comment]
Precise: https://ocd.appscale.com:8080/job/Daily%20Build/1063/
Trusty: https://ocd.appscale.com:8080/job/Daily%20Build/1067/

**Pull request merged**